ce/api/percentileanomaly.py

```diff
@@ -183,62 +291,66 @@ def canonical_timestamp(timestamp, timescale, timeidx):
                                                             row['units']))
 
                     if row['climatology'] == climatology:
-                        add_to_nested_dict(projected_data,
-                                           row["mean"],
-                                           row["timescale"],
-                                           ctimestamp,
-                                           row["model"])
+
+                        projected_data = add_to_nested_li(projected_data, row, ctimestamp)
+
                     elif (calculate_anomaly and
                             row['model'] == baseline_model and
                             row['climatology'] == baseline_climatology):
-                        add_to_nested_dict(baseline_data, row['mean'],
-                                           row['timescale'], ctimestamp)
+                        obj = create_data_object(row["mean"], row["timescale"], ctimestamp)
+                        idx = generate_list_idx(row["timescale"], row["timeidx"])
+                        baseline_data[idx] = obj
+
+
+        projected_data = [data_object for data_object in projected_data if data_object != None]
+        baseline_data = [data_object for data_object in baseline_data if data_object != None]
 
         # calculate percentiles and anomalies
```

Comment:
This chunk of code (the entire for loop) is long and hard to understand. I think we might do better to break out the part that determines the baseline as a separate function that is called inside the loop, leaving the remainder as-is. It would also help my tiny brain if there were some comments explaining the baseline calculation a bit, specifically the conditions under which a baseline found. These would naturally go in the baseline function.